pkgs/build-support/mkshell/minimal.nix

```diff
@@ -0,0 +1,60 @@
+{ writeTextFile, writeScript, system }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell. This differs from the traditional `mkShell` in that it is
+# ultra minimalist in dependencies.
+# It does not come with traditional stdenv (i.e. coreutils).
+# Its only dependency is essentially bash.
```

Comment:
This can be simplified a lot. There's no point beyond ultra and nothing is smaller than minimal.

It's better to describe the few things it does do.

Reply:
Let's move this alongside documentation to follow-up ?